forui/lib/src/widgets/text_field/text_field.dart

```diff
@@ -0,0 +1,691 @@
+  /// [DragStartBehavior.down] will make drag behavior feel slightly more reactive.
+  ///
+  /// See also:
+  ///  * [DragGestureRecognizer.dragStartBehavior], which gives an example for the different behaviors.
+  final DragStartBehavior dragStartBehavior;
+
+  // TODO: MouseCursor? mouseCursor;
+
+  // TODO: InputCounterWidgetBuilder? buildCounter;
+
+  /// The [ScrollPhysics] to use when vertically scrolling the input. If not specified, it will behave according to the
+  /// current platform.
+  ///
+  /// See [Scrollable.physics].
+  final ScrollPhysics? scrollPhysics;
+
+  /// The [ScrollController] to use when vertically scrolling the input. If null, it will instantiate a new ScrollController.
+  ///
+  /// See [Scrollable.controller].
+  final ScrollController? scrollController;
+
+  /// A list of strings that helps the autofill service identify the type of this text input.
+  ///
+  /// See [TextField.autofillHints] for more information.
+  final Iterable<String>? autofillHints;
+
+  /// Restoration ID to save and restore the state of the text field.
+  ///
+  /// See [TextField.restorationId] for more information.
+  final String? restorationId;
+
+  /// Whether iOS 14 Scribble features are enabled for this widget. Defaults to true.
+  ///
+  /// Only available on iPads.
+  final bool scribbleEnabled;
+
+  /// Whether to enable that the IME update personalized data such as typing history and user dictionary data.
+  ///
+  /// See [TextField.enableIMEPersonalizedLearning] for more information.
+  final bool enableIMEPersonalizedLearning;
+
+  // TODO: ContentInsertionConfiguration? contentInsertionConfiguration
+
+  /// Builds the text selection toolbar when requested by the user.
+  ///
+  /// See [TextField.contextMenuBuilder] for more information.
+  final EditableTextContextMenuBuilder? contextMenuBuilder;
+
+  /// Determine whether this text field can request the primary focus.
+  ///
+  /// Defaults to true. If false, the text field will not request focus when tapped, or when its context menu is
+  /// displayed. If false it will not be possible to move the focus to the text field with tab key.
+  final bool canRequestFocus;
+
+  /// Controls the undo state.
+  ///
+  /// If null, this widget will create its own [UndoHistoryController].
+  final UndoHistoryController? undoController;
+
+  /// Configuration that details how spell check should be performed.
+  ///
+  /// Specifies the [SpellCheckService] used to spell check text input and the [TextStyle] used to style text with
+  /// misspelled words.
+  ///
+  /// If the [SpellCheckService] is left null, spell check is disabled by default unless the [DefaultSpellCheckService]
+  /// is supported, in which case it is used. It is currently supported only on Android and iOS.
+  ///
+  /// If this configuration is left null, then spell check is disabled by default.
+  final SpellCheckConfiguration? spellCheckConfiguration;
+
+  /// The label.
+  final String? label;
```

Comment:
Since we're providing a label, does it make sense to include a field for an error message?

Reply:
I was thinking of addressing it in a future PR since we didn't have the error colors sorted out.